src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,369 @@
+  }
+
+
+  private Command getVelocitiesCheckCommand() {
+    return Commands.parallel(
+      Commands.run(this::runIntakeStateMachine),
+      Commands.waitSeconds(1).andThen(
+        Commands.runOnce(() -> {
+          for (int i = 0; i < 4; i++) {
+            checkMotorVelocity(intakeMotors[i]);
+          }
+        })
+      )
+    );
+  }
+
+  private void checkMotorVelocity(IntakeMotor motor) {
+    if (motor == IntakeMotor.RIGHT_ROLLER) {
+      if (inputs.rightRollerVelocityRotationsPerSecond 
+      - inputs.rightRollerReferenceVelocityRPS 
+      > IntakeConstants.ROLLER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Right Roller Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.LEFT_ROLLER) {
+      if (inputs.leftRollerVelocityRotationsPerSecond 
+      - inputs.leftRollerReferenceVelocityRPS 
+      > IntakeConstants.ROLLER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Left Roller Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.DRUM) {
+      if (inputs.drumVelocityRotationsPerSecond 
+      - inputs.drumReferenceVelocityRPS 
+      > IntakeConstants.DRUM_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Drum Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.KICKER) {
+      if (inputs.kickerVelocityRotationsPerSecond 
+      - inputs.kickerReferenceVelocityRPS 
+      > IntakeConstants.KICKER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Kicker Too Slow"
+        );
+      }
+    }
+  }
+
+  private Command getStatesCheckCommand() {
+    IntakeState[] desiredStateSequence = {
+      IntakeState.EMPTY,
+      IntakeState.NOTE_IN_INTAKE,
+      IntakeState.IN_BETWEEN_INTAKE_AND_DRUM,
+      IntakeState.NOTE_IN_DRUM,
+      IntakeState.IN_BETWEEN_DRUM_AND_KICKER,
+      IntakeState.NOTE_IN_KICKER
+    };
+    
+    ArrayList<IntakeState> actualStateSeqeunce = new ArrayList<>();
+
+    return Commands.parallel(
+      Commands.run(this::runIntakeStateMachine),
+      Commands.waitSeconds(1).andThen(
+        Commands.runOnce(() -> checkStateSequence(actualStateSeqeunce, desiredStateSequence))
+      )
+    );
+  }
+
+  private void checkStateSequence(ArrayList<IntakeState> actualStateSequence, IntakeState[] desiredStateSequence) {
+    actualStateSequence.add(IntakeState.EMPTY);
+    IntakeState mostRecentIntakeState = IntakeState.EMPTY;
+
+    boolean correct = actualStateSequence.get(0).equals(desiredStateSequence[0]);
+
+    while (correct) {
+      if (intakeState != mostRecentIntakeState) {
+        actualStateSequence.add(intakeState);
+        mostRecentIntakeState = intakeState;
+      }
+
+      for (int i = 0; i < actualStateSequence.size(); i++) {
```

Comment:
Maybe check if the actual state matches the expected state in the above if statement instead of a loop? This is a bit tricky. We can figure it out together.

Reply:
I think I would need to have an instance variable for the amount of times that the state has been changed based off of this (would reset it at the start of each systems check). I can implement this pretty quickly but will keep the loop for now until we talk.